### ISSUE TITLE
Allow Memory Hard Limit setting.

### ISSUE THREAD

**Issue by h5aaimtron** [aws/copilot-cli#4791]
Per the article:

https://aws.amazon.com/blogs/developer/configuring-net-garbage-collection-for-amazon-ecs-and-aws-lambda/

.NET GC sees the underlying hosts' total memory instead of the containers memory which can lead to OOM errors. The article states that you can specify the hard limit of the container which solves the issue, however; I haven't seen an easy to implement mechanism via manifest file. I'm not sure if this is implemented, but I could see it being very useful.

**Comment by bvtujo** [comment]
Hi there! You're right that there's not a native way to do this, but you can do it pretty easily using `copilot svc override --tool yamlpatch`

Then, in `copilot/web/overrides/cfn.patches.yml`, add the following code:

```yaml
- op: add
  path: /Resources/TaskDefinition/Properties/ContainerDefinitions/0/Memory
  value: ${MEMORY_LIMIT}
```

A quick copilot deploy and you're done. You can supply the `--diff` flag to see the changes before you deploy.

**Comment by h5aaimtron** [comment]
@bvtujo I get an error when I run that command and select yamlpatch.
read dir "overrides/yamlpatch": open templates\overrides\yamlpatch: file does not exist

**Comment by bvtujo** [comment]
Apologies, the command I pasted above appears to have been corrupted in the process. 

You ran:
```
copilot svc override --tool yamlpatch
```
Correct?

If so, and you still ran into this problem, we have something else going on. 

What version of Copilot are you using? Overrides was only released last month, in 1.27.0.

**Comment by h5aaimtron** [comment]
@bvtujo yes that is the command I ran. I'm using Copilot 1.27, however; the app and svc were originally created with 1.26.

**Comment by bvtujo** [comment]
Are you running windows, by any chance? I'm unable to reproduce this on macOS or linux unfortunately.

**Comment by h5aaimtron** [comment]
@bvtujo 
Yes I am.

**Comment by Lou1415926** [comment]
Hey @h5aaimtron ! Tried on my Windows machine and can confirm that this is a bug specifically on Windows. Sorry for the trouble :( ! I've sent out a fix which should be rolled out in the next release!

**Comment by Lou1415926** [comment]
Oh hey @h5aaimtron!  Before we release the fix, you can skip the `copilot svc override` command and just run ⬇️
```
mkdir ./copilot/<svc>/overrides && type nul > ./copilot/<svc>/overrides/cfn.patches.yml
```
Then you can edit `cfn.patches` and run `copilot svc deploy`.
Thanks @efekarakus  and @dannyrandall for the workaround!

**Comment by zeeshanpolaris** [comment]
When will this fix release? I am on windows and stuck with not be able to add ingress source_ips or additional security groups for a private load balancer natively so I need to override. I need to whitelist more private ips that are beyond the vpc and subnet cidr ranges.

**Comment by Lou1415926** [comment]
@zeeshanpolaris The fix will be released ASAP 🙇🏼 Apologies on the delay, and apologies that I am not able to give you a specific date, but I believe it will be pretty soon. I will keep you posted once the release is out. Sorry for the trouble : (

**Comment by zeeshanpolaris** [comment]
@Lou1415926 Thank you. I got around to this by switching over to my mac and running commands there and copying the results of cdk and yml back over to windows.

**Comment by Lou1415926** [comment]
Hello @zeeshanpolaris and @h5aaimtron ! The fix is now released in [v1.28.0](https://github.com/aws/copilot-cli/releases/tag/v1.28.0) 🎊!

**Comment by zeeshanpolaris** [comment]
woohoo, thank you so much
